test/infrastructure/docker/e2e/docker_test.go

```diff
@@ -213,6 +215,20 @@ var _ = Describe("Docker", func() {
 					}
 					return upgraded, nil
 				}, "10m", "30s").Should(Equal(int(*controlPlane.Spec.Replicas)))
+
+				Eventually(func() (bool, error) {
+					ds := &appsv1.DaemonSet{}
+
+					if err := workloadClient.Get(ctx, ctrlclient.ObjectKey{Name: "kube-proxy", Namespace: metav1.NamespaceSystem}, ds); err != nil {
+						return false, err
+					}
+					if ds.Spec.Template.Spec.Containers[0].Image == "k8s.gcr.io/kube-proxy:v1.17.2" {
```

Comment:
Might be good to keep the version as a variable within this Specify so that it isn't repeated here and on L191, WDYT? Or could take the value from the `controlPlane` and then it won't ever drift if the test gets updated 🤔 

```suggestion
					if ds.Spec.Template.Spec.Containers[0].Image == fmt.Sprintf("k8s.gcr.io/kube-proxy:%s", controlPlane.Spec.Version) {
```

Reply:
Good point! The rest of the tests have it hardcoded. I will fix that in a followup PR.